fairing/ml_tasks/tasks.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+import fairing
+from fairing.deployers.job.job import Job
+from fairing.deployers.serving.serving import Serving
+from .utils import guess_preprocessor, guess_builder
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTask:
+    """
+    Base class for handling high level ML tasks.
+    args:
+        entry_point: An object or reference to the source code that has to be deployed.
+        base_docker_image: Name of the base docker image that should be used as a base image
+            when building a new docker image as part of an ML task deployment.
+        docker_registry: Docker registry to store output docker images.
+        input_files: list of files that needs to be packaged along with the entry point.
+            E.g. local python modules, trained model weigths, etc.
+    """
+    def __init__(self, entry_point, base_docker_image, docker_registry, input_files=[]):
```

Comment:
`input_files` should be None instead of an empty list -- mutable default args will persist across method calls.

Reply:
Ack. Good catch.